### PR TITLE
Fix exception when routing to company

### PR DESCRIPTION
**Pull request by achohbee**
Fix exception that gets thrown when trying to route to the company.

### CODE REVIEW

**Comment by achohbee** [comment]
This is a quick and fast kludge, there might be a better fix for an underlying problem. Also untested at the moment, I can muck around getting this to build local for me later to test.

The callstack that gets sent to terminal whenever you try to hit enter after typing in `company` is:
```
LethalLevelLoader.TerminalManager.LockedNodeEventTest (TerminalNode& currentNode, TerminalNode& loadNode) (at <bed140a3e66641339c3af806fdbbd3ed>:IL_0006)
LethalLevelLoader.TerminalManager.OnBeforeLoadNewNode (TerminalNode& node) (at <bed140a3e66641339c3af806fdbbd3ed>:IL_004D)
LethalLevelLoader.Patches.TerminalLoadNewNode_Prefix (Terminal __instance, TerminalNode& node) (at <bed140a3e66641339c3af806fdbbd3ed>:IL_001B)
(wrapper dynamic-method) Terminal.DMD<Terminal::LoadNewNode>(Terminal,TerminalNode)
(wrapper dynamic-method) Terminal.DMD<Terminal::OnSubmit>(Terminal)
UnityEngine.Events.InvokableCall.Invoke () (at <e27997765c1848b09d8073e5d642717a>:IL_0010)
UnityEngine.Events.UnityEvent`1[T0].Invoke (T0 arg0) (at <e27997765c1848b09d8073e5d642717a>:IL_0049)
TMPro.TMP_InputField.SendOnEndEdit () (at <9baa8119dbcd498f902c1a92f10843c1>:IL_000E)
TMPro.TMP_InputField.ReleaseSelection () (at <9baa8119dbcd498f902c1a92f10843c1>:IL_001B)
TMPro.TMP_InputField.DeactivateInputField (System.Boolean clearSelection) (at <9baa8119dbcd498f902c1a92f10843c1>:IL_00B0)
TMPro.TMP_InputField.OnUpdateSelected (UnityEngine.EventSystems.BaseEventData eventData) (at <9baa8119dbcd498f902c1a92f10843c1>:IL_007C)
UnityEngine.EventSystems.ExecuteEvents.Execute (UnityEngine.EventSystems.IUpdateSelectedHandler handler, UnityEngine.EventSystems.BaseEventData eventData) (at <17de86d66286445fa95af59fa8fa644d>:IL_0000)
UnityEngine.EventSystems.ExecuteEvents.Execute[T] (UnityEngine.GameObject target, UnityEngine.EventSystems.BaseEventData eventData, UnityEngine.EventSystems.ExecuteEvents+EventFunction`1[T1] functor) (at <17de86d66286445fa95af59fa8fa644d>:IL_0067)
UnityEngine.EventSystems.EventSystem:Update()
```

File: LethalLevelLoader/Patches/TerminalManager.cs [hunk 1]
```diff
@@ -81,6 +81,11 @@ internal static void CacheTerminalReferences()
 
         internal static bool LockedNodeEventTest(ref TerminalNode currentNode, ref TerminalNode loadNode)
         {
+            // This prevents an IndexOutOfRangeException when travelling to the company store
+            if (StartOfRound.Instance.levels.Length <= loadNode.displayPlanetInfo) {
```
Comment (IHaveNoNameYetIDo) [inline LethalLevelLoader/Patches/TerminalManager.cs:85]: I also looked into this and my guess is that the exact reason is that for the "route company" TerminalNode displayPlanetInfo is set to its default value -1. 
Even if you look at the game code-base on Terminal::TextPostProcess you can see they are checking against displayPlanetInfo != -1

I patched the IL (look at the IL attached below) for the affected method in the latest DLL version of the mod and after testing it, seems to work:
```asm
    ldarg.1
    ldind.ref
    ldfld int32 ['Assembly-CSharp']TerminalNode::displayPlanetInfo
    ldc.i4.m1  // Push -1 onto the stack
    bne.un.s original  // Branch if not equal to -1
    ldc.i4.1  // Push true onto the stack
    ret  // Return true
    ···
    original:
    // Original code starts here
```
```suggestion
            if (loadNode.displayPlanetInfo == -1) {
```

Comment (achohbee) [reply LethalLevelLoader/Patches/TerminalManager.cs]: Good catch! The -1 case occurred to me later, but I didn't go back to update the PR. My group went with the suggested other mod that masks this problem as a stopgap workaround.

**Comment by IAmBatby** [comment]
Closing, Fixed in v1.2.0.3

**Pull request closed without merging**